src/Avalonia.Diagnostics/Diagnostics/ViewModels/MainViewModel.cs

```diff
@@ -185,5 +208,75 @@ private void KeyboardPropertyChanged(object sender, PropertyChangedEventArgs e)
             }
         }
+
+        [DependsOn(nameof(TreePageViewModel.SelectedNode))]
+        [DependsOn(nameof(Content))]
+        bool CanShot(object paramter)
+        {
+            return Content is TreePageViewModel tree
+                && tree.SelectedNode != null
+                && tree.SelectedNode.Visual != null
+                && tree.SelectedNode.Visual.VisualRoot != null;
+        }
+
+        void Shot(object parameter)
+        {
+            // This is a workaround because MethodToCommand does not support the asynchronous method.
+            Task.Factory.StartNew(arg =>
+                {
+                    if (arg is IControl control)
+                    {
+                        try
+                        {
+                            var folder = GetScreenShotDirectory(_root);
+                            if (System.IO.Directory.Exists(folder) == false)
+                            {
+                                System.IO.Directory.CreateDirectory(folder);
+                            }
+                            var filePath = System.IO.Path.Combine(folder
+                                , $"{DateTime.Now:yyyyMMddhhmmssfff}.png");
+
+                            var output = new System.IO.FileStream(filePath, System.IO.FileMode.Create);
+                            Dispatcher.UIThread.Post(() =>
+                                {
+                                    control.RenderTo(output);
+                                    output.Dispose();
+                                }
+                            );
+
+                        }
+                        catch (Exception ex)
+                        {
+                            System.Diagnostics.Debug.WriteLine(ex.Message);
+                            //TODO: Notify error
+                        }
+
+                    }
+
+
+                }, (Content as TreePageViewModel)?.SelectedNode?.Visual);
+        }
+
+        /// <summary>
+        /// Return the path of the screenshot folder according to the rules indicated in issue <see href="https://github.com/AvaloniaUI/Avalonia/issues/4743">GH-4743</see>
+        /// </summary>
+        /// <param name="root"></param>
+        /// <returns></returns>
+        string GetScreenShotDirectory(TopLevel root)
+        {
+            var rootType = root.GetType();
+            var windowName = rootType.Name;
+            var assembly = Assembly.GetExecutingAssembly();
+            var appName = Application.Current?.Name
+                ?? assembly.GetCustomAttribute<AssemblyProductAttribute>()?.Product
+                ?? assembly.GetName().Name;
+            var appVerions = assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion
+                ?? assembly.GetCustomAttribute<AssemblyVersionAttribute>().Version;
+            return System.IO.Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.MyPictures, Environment.SpecialFolderOption.Create)
```

Comment:
There should be ability to set custom folder.
Probably as an parameter in Attach method (it should have some options param though).